internal/summon/command/run.go

```diff
@@ -25,33 +27,83 @@ type Subcommand struct {
 
 // buildEnvironment builds the environment strings from the map of secrets values, along with the
 // secrets configuration metadata and the temp files location.
-func buildEnvironment(secrets map[string]string, secretsMap secretsyml.SecretsMap, tempFactory *TempFactory) (env []string, err error) {
-	env = make([]string, len(secrets))
-	for key, value := range secrets {
-		envvar := formatForEnv(key, value, secretsMap[key], tempFactory)
+func buildEnvironment(
+	secrets map[string]string,
+	secretsMap secretsyml.SecretsMap,
+	tempFactory *TempFactory,
+) ([]string, error) {
+	env := make([]string, 0, len(secrets))
+	keys := make([]string, 0, len(secrets))
+
+	for k := range secrets {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for _, key := range keys {
+		envvar, err := formatForEnv(key, secrets[key], secretsMap[key], tempFactory)
+		if err != nil {
+			return nil, err
+		}
 		env = append(env, envvar)
 	}
-	return
+
+	return env, nil
 }
 
 // resolveSecrets obtains the value of each requested secret.
-func resolveSecrets(provider plugin_v1.Provider, secretsMap secretsyml.SecretsMap) (result map[string]string, err error) {
-	result = make(map[string]string)
+func resolveSecrets(provider plugin_v1.Provider, secretsMap secretsyml.SecretsMap) (map[string]string, error) {
+	var err error
+	result := make(map[string]string)
+
+	var varSecretsSpecKeys []string
+	var varSecretsSpecPaths []string
+
 	for key, spec := range secretsMap {
-		var value string
 		if spec.IsVar() {
-			var valueBytes []byte
-			if valueBytes, err = provider.GetValue(spec.Path); err != nil {
-				return
-			}
-			value = string(valueBytes)
+			varSecretsSpecKeys = append(varSecretsSpecKeys, key)
+			varSecretsSpecPaths = append(varSecretsSpecPaths, spec.Path)
 		} else {
 			// If the spec isn't a variable, use its value as-is
-			value = spec.Path
+			value := spec.Path
+			result[key] = value
 		}
-		result[key] = value
 	}
-	return
+
+	if atLeastOneVar := len(varSecretsSpecPaths) > 0; !atLeastOneVar {
```

Comment:
Not a blocker but this could have been simpler with just the condition:
```go
# If there are no variables to resolve, return what we have
if len(varSecretsSpecPaths) == 0 {
```